core/src/main/java/org/everit/json/schema/ValidatingVisitor.java

```diff
@@ -46,13 +48,18 @@ void visit(Schema schema) {
         super.visit(schema);
     }
 
-    ValidatingVisitor(Object subject, ValidationFailureReporter failureReporter, ReadWriteValidator readWriteValidator) {
+    ValidatingVisitor(Object subject, ValidationFailureReporter failureReporter, ReadWriteValidator readWriteValidator, SchemaVisitorListener schemaVisitorListener) {
         if (subject != null && !VALIDATED_TYPES.stream().anyMatch(type -> type.isAssignableFrom(subject.getClass()))) {
             throw new IllegalArgumentException(format(TYPE_FAILURE_MSG, subject.getClass().getSimpleName()));
         }
         this.subject = subject;
         this.failureReporter = failureReporter;
         this.readWriteValidator = readWriteValidator;
+        this.schemaVisitorListener = schemaVisitorListener;
+    }
+
+    ValidatingVisitor(Object subject, ValidationFailureReporter failureReporter, ReadWriteValidator readWriteValidator) {
```

Comment:
why do we need this constructor?

Reply:
To keep compatibility with previous versions and people not using the SchemaVisitorListener. Since there are no default arguments in java I was keeping it.

Reply:
Cool! I like removing things!